packages/mui-utils/src/useUndoAndRedo.ts

```diff
@@ -0,0 +1,55 @@
+import isEqual from 'lodash/isEqual';
+import { useState } from 'react';
+
+const useUndoAndRedo = <T>(initialValue?: T) => {
```

Comment:
We may want to add an option about how many steps in the history we want to keep.